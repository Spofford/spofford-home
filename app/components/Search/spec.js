import React from "react"
import expect from "expect"
import { shallow } from "enzyme"

import { Search } from "./"

describe("<Search />", () => {
  it("should render", () => {
    const renderedComponent = shallow(
      <Search />
    )
    expect(renderedComponent.is("div")).toEqual(true)
  })
})
