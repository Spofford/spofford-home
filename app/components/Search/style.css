.search .body-container {
  padding-bottom:50px;
}
