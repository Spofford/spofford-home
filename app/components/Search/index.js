import React from "react"
const ReactMarkdown = require('react-markdown')
import * as contentful from 'contentful'
import style from "./style.css"
import { Link } from 'react-router-dom'
import FontAwesome from "react-fontawesome";

import { default as Header } from "../Header"

const client = contentful.createClient({
  space: 'cahjy08ew1qz',
  accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
})


export class Search extends React.Component {
  state = {
    query: this.props.match.params.query,
    posts: []
  }


  componentDidMount() {
    var self = this
    client.getEntries({
      content_type: 'blogPost',
      'query': self.state.query
    })
    .then((response) => this.setState({posts:response.items}))
    .catch(console.error)
  }

  render() {
    const myData = [].concat(this.state.posts)
      .sort((a, b) => a.fields.datePublished < b.fields.datePublished)
      .map((item, i) =>
          <div className="postContainer" key={i}>
            <img src={item.fields.primaryImage.fields.file.url} />
            <h2>{item.fields.head}</h2>
            <h3>{item.fields.subhead}</h3>
            <Link to={'/post/' + item.sys.id}>Read More</Link>
          </div>
      );

    return (
      <div className="search">
        <Header />
        <div className="body-container">
          <div className="posts">
            <h2>Search results for "{this.state.query}"</h2>
            {myData}
            <Link to='/'><FontAwesome name='chevron-left' />Return to home</Link>
          </div>
        </div>
      </div>
    )
  }
}

export default Search
