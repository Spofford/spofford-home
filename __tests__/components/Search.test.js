import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'
import * as contentful from 'contentful'

import Search from '../../app/components/Search';

describe('<Search />', () => {
  describe('render()', () => {
    var client = contentful.createClient({
      space: 'cahjy08ew1qz',
      accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
    });

    test('renders the component', () => {

        var match = {
          params: {
            entity: "search query"
          }
        }
        const wrapper = shallow(<Search match={match} />);

        expect(toJson(wrapper)).toMatchSnapshot();
      })
    // Test for contentful content

  });
});
