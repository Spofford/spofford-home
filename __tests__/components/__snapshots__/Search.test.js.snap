// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> render() renders the component 1`] = `
<div
  className="search"
>
  <Connect(Header) />
  <div
    className="body-container"
  >
    <div
      className="posts"
    >
      <h2>
        Search results for "
        "
      </h2>
      <Link
        replace={false}
        to="/"
      >
        <FontAwesome
          name="chevron-left"
        />
        Return to home
      </Link>
    </div>
  </div>
</div>
`;
